### PR TITLE
Mlhr 1824 ram3

### PR DESCRIPTION
**Pull request by amberarrow**
Add app data support as a new app and restore state of original app.

### CODE REVIEW

File: demos/pi/src/main/java/com/datatorrent/demos/pi/ApplicationAppData.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    RandomEventGenerator rand = dag.addOperator("rand", new RandomEventGenerator());
+    PiCalculateOperator calc = dag.addOperator("picalc", new PiCalculateOperator());
+
+
+    dag.addStream("rand_calc", rand.integer_data, calc.input).setLocality(locality);
+
+    String gatewayAddress = dag.getValue(DAG.GATEWAY_CONNECT_ADDRESS);
+
+    if (StringUtils.isEmpty(gatewayAddress)) {
+      throw new RuntimeException("Error: No GATEWAY_CONNECT_ADDRESS");
```
Comment (ilooner) [inline demos/pi/src/main/java/com/datatorrent/demos/pi/ApplicationAppData.java]: I think running without a gateway connect address is still valid. The application should just output to the console and the app data operators should not be connected.

Comment (ilooner) [inline demos/pi/src/main/java/com/datatorrent/demos/pi/ApplicationAppData.java]: Nevermind, i see that the other application serves this purpose.

File: demos/pi/src/main/resources/META-INF/properties.xml [hunk 2]
```diff
@@ -44,6 +44,32 @@
     <value>PiDemoQueryResult</value>
```
Comment (ilooner) [inline demos/pi/src/main/resources/META-INF/properties.xml]: This property is no longer needed.

**Comment by amberarrow** [comment]
Removed unneeded properties.

**Comment by ilooner** [comment]
Hey @amberarrow is this still good to merge? If so I will merge it today.

**Comment by amberarrow** [comment]
Yes, please merge.

**Comment by tweise** [comment]
@amberarrow Is this PR still relevant? It will need rebase.

**Comment by amberarrow** [comment]
No longer relevant, closing.

**Pull request closed without merging**